Update tests to work under CMake

tobiasvanelk opened the pull request:
Updated tests to work under CMake.

Modified some test cases to use a global variable pointing to the test file directory and using a global fixture to initialize it.

kees-jan commented on plugins/sep/test/main.cc:
```diff
@@ -1,3 +1,27 @@
+    }
+};
+// Register the fixture with the test module
+BOOST_GLOBAL_FIXTURE(F);
```
I really like the global fixture. However, it is currently used to initialize a global variable located in `constants.hh`, that isn't actually constant.

- It should be possible to make `testFileDir` a member of the global fixture, and then use it from there, instead of using the global variable.
- Having done that, we probably should rename the fixture to something like `TestFiles`
- Perhaps it should have a method `std::string getPathToFile(std::string filename)`, to replace the `(testFileDir / "whatever.sli").toString()` that is now sprinkled across the tests.

kees-jan commented:
Currently, the autoconf build on travis-ci is red. Could you please add `constants.cc` and `constants.hh` to `plugins_sep_test_sep_tests_SOURCES` in `Makefile.am`? Once that is done and `travis-ci` reports green again, I'll happily merge this, provided you promise to address my other remarks in an upcoming pull-request.

tobiasvanelk commented:
> Currently, the autoconf build on travis-ci is red. Could you please add `constants.cc` and `constants.hh` to `plugins_sep_test_sep_tests_SOURCES` in `Makefile.am`? Once that is done and `travis-ci` reports green again, I'll happily merge this, provided you promise to address my other remarks in an upcoming pull-request.

I added the files to the Makefile.am and removed one an unnecessary variable the static analysis tool was unhappy about. Hopefully it will pass all travis-ci checks now. I'll be making a pull request later today for the other feedback such as incorporating the path variable as a member of the fixture.
